Minio/Helper/s3utils.cs

```diff
@@ -80,7 +83,52 @@ internal static string GetPath(string p1, string p2)
 
         }
 
-      
-     
+        // IsValidDomain validates if input string is a valid domain name.
+        public static bool IsValidDomain(string host)
+        {
+            // See RFC 1035, RFC 3696.
+            host = host.Trim();
+            if ((host.Length == 0) || (host.Length > 255))
+            {
+                return false;
+            }
+            // host cannot start or end with "-"
+            if ((host.Substring(host.Length - 1) == "-") || (host.Substring(1) == "-"))
+            {
+                return false;
+            }
+            // host cannot start or end with "_"
+            if ((host.Substring(host.Length - 1) == "_") || (host.Substring(1) == "_"))
+            {
+                return false;
+            }
+            // host cannot start or end with a "."
+            if ((host.Substring(host.Length - 1) == ".") || (host.Substring(1) == "."))
+            {
+                return false;
+            }
+            return Uri.CheckHostName(host) != UriHostNameType.Unknown;
+            // No need to regexp match, since the list is non-exhaustive.
+            // We let it valid and fail later.
+        }
+
+        // IsValidIP parses input string for ip address validity.
+        public static bool IsValidIP(string ip)
```

Comment:
Aren't we supporting IPv6 as well?  No standard library for checking valid IPv4 and IPv6 addresses?